ballerina/tests/json_to_json_test.bal

```diff
@@ -0,0 +1,517 @@
+    string city;
+    int id;
+};
+
+type RN record {|
+    int id;
+    string name;
+    AddressN address;
+|};
+
+@test:Config {
+    groups: ["jsonToJson"]
+}
+isolated function testJsonToJson13() returns error? {
+    // json jsonContent = {
+    //             "id": 12,
+    //             "name": "Anne",
+    //             "address": {
+    //                 "street": "Main",
+    //                 "city": "94",
+    //                 "id": true
+    //             }
+    //         };
+
+    string strContent = "{\n\"id\": 12,\n\"name\": \"Anne\",\n\"address\": {\n\"street\": \"Main\",\n\"city\": \"94\",\n\"id\": true\n}\n}";
+
+    byte[] bytes = strContent.toBytes();
+
+    RN|Error x = fromJsonByteArrayWithType(bytes, RN);
+    test:assertTrue(x is error);
+    test:assertEquals((<error>x).message(), "incompatible value 'true' for type 'int' in field 'address.id' at line: 8 column: 0");
+}
+
+type RN2 record {|
+    int id;
+    string name;
+|};
+
+@test:Config {
+    groups: ["jsonToJson"]
+}
+isolated function testJsonToJson14() returns error? {
+    json jsonContent = {
+                "id": 12
+            };
+
+    byte[] bytes = jsonContent.toString().toBytes();
+
+    RN2|Error x = fromJsonByteArrayWithType(bytes, RN2);
+    test:assertTrue(x is error);
+    test:assertEquals((<error>x).message(), "required field 'name' not present in JSON at line: 1 column: 10");
+}
+
+@test:Config {
+    groups: ["jsonToJson"]
+}
+isolated function testJsonToJson15() returns error? {
+    json jsonContent = {
+                "id": 12,
+                "name": "Anne",
+                "address": {
+                    "street": "Main",
+                    "city": "94"
+                }
+            };
+
+    byte[] bytes = jsonContent.toString().toBytes();
+
+    RN|Error x = fromJsonByteArrayWithType(bytes, RN);
+    test:assertTrue(x is error);
+    test:assertEquals((<error>x).message(), "required field 'id' not present in JSON at line: 1 column: 63");
+}
+
+
+@test:Config {
+    groups: ["jsonToJson"]
+}
+isolated function testJsonToJson16() returns error? {
+    json jsonContent = {
+        "street": "Main",
+        "city": "Mahar",
+        "house": [94, [1, 3, "4"]]
+    };
+    byte[] bytes = jsonContent.toString().toBytes();
+
+    TestArr3|error x = fromJsonByteArrayWithType(bytes, TestArr3);
+    test:assertTrue(x is error);
+}
+
+@test:Config {
+    groups: ["jsonToJson"]
+}
+isolated function testJsonToJson17() returns error? {
+    json jsonContent = {
+                "id": 12,
+                "name": "Anne",
+                "address": {
+                    "id": 34,
+                    "city": "94"
+                }
+            };
+
+    byte[] bytes = jsonContent.toString().toBytes();
+
+    RN|Error x = fromJsonByteArrayWithType(bytes, RN);
+    test:assertTrue(x is error);
+    test:assertEquals((<error>x).message(), "required field 'address.street' not present in JSON at line: 1 column: 56");
+}
+
```

Comment:
Do we have any test cases covering the projection? Most of the tests are covering the exact match for open records and closed records.

Reply:
We have some cases, will add more.